libnetwork/iptables/iptables.go

```diff
@@ -88,54 +88,40 @@ func (e ChainError) Error() string {
 	return fmt.Sprintf("Error iptables %s: %s", e.Chain, string(e.Output))
 }
 
-func probe() {
-	path, err := exec.LookPath("iptables")
-	if err != nil {
-		logrus.Warnf("Failed to find iptables: %v", err)
-		return
-	}
-	if out, err := exec.Command(path, "--wait", "-t", "nat", "-L", "-n").CombinedOutput(); err != nil {
-		logrus.Warnf("Running iptables --wait -t nat -L -n failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
-	}
-	_, err = exec.LookPath("ip6tables")
-	if err != nil {
-		logrus.Warnf("Failed to find ip6tables: %v", err)
-		return
-	}
-}
-
-func initFirewalld() {
-	if err := FirewalldInit(); err != nil {
-		logrus.Debugf("Fail to initialize firewalld: %v, using raw iptables instead", err)
-	}
-}
-
 func detectIptables() {
 	path, err := exec.LookPath("iptables")
 	if err != nil {
+		logrus.Warnf("Failed to find iptables: %v", err)
 		return
 	}
 	iptablesPath = path
 
-	supportsXlock = exec.Command(iptablesPath, "--wait", "-L", "-n").Run() == nil
+	supportsXlock = true
+	if out, err := exec.Command(path, "--wait", "-L", "-n").CombinedOutput(); err != nil {
+		logrus.Warnf("Running iptables --wait -L -n failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
+	}
 	mj, mn, mc, err := GetVersion()
 	if err != nil {
 		logrus.Warnf("Failed to read iptables version: %v", err)
-		return
 	}
 	supportsCOpt = supportsCOption(mj, mn, mc)
 
 	path, err = exec.LookPath("ip6tables")
 	if err != nil {
-		return
+		logrus.Warnf("Failed to find ip6tables: %v", err)
 	}
 	ip6tablesPath = path
 }
 
+func initFirewalld() {
+	if err := FirewalldInit(); err != nil {
+		logrus.Debugf("Fail to initialize firewalld: %v, using raw iptables instead", err)
```

Comment:
While we're changing this; perhaps also use `WithError()` for this one, and I tend to prefer avoiding use of `Fail` / `Failed` in non-fatal situations (it can be confusing to see a "failed" message in the logs). Something like "unable to" may be a better fit;

```go
logrus.WithError(err).Debug("unable to initialize firewalld; using raw iptables instead")
```